Sources/AccountSDKIOSWeb/Client/Client.swift

```diff
@@ -168,14 +186,19 @@ public class Client: CustomStringConvertible {
             preconditionFailure("Couldn't create loginURL")
         }
         
+        tracker?.interaction(.view, with: .webBrowser, additionalFields: [.getLoginSession(withMFA), .loginHint(loginHint), .withAssertion((assertion != nil) ? true : false), .extraScopeValues(extraScopeValues)])
+        
         let session = ASWebAuthenticationSession(url: url, callbackURLScheme: clientScheme) { callbackURL, error in
             guard let url = callbackURL else {
                 if case ASWebAuthenticationSessionError.canceledLogin? = error {
                     SchibstedAccountLogger.instance.debug("Login flow was cancelled")
+                    self.tracker?.engagement(.click(on: .cancel), in: .webBrowser)
                     completion(.failure(.canceled))
                 } else {
                     SchibstedAccountLogger.instance.error("Login flow error: \(String(describing: error))")
-                    completion(.failure(.unexpectedError(message: "ASWebAuthenticationSession failed: \(String(describing: error))")))
+                    let error = LoginError.unexpectedError(message: "ASWebAuthenticationSession failed: \(String(describing: error))")
```

Comment:
`String(describing: error)` will always print `Optional<String>(value)` even if it's not nill. Let's please not have that in the analytics. 

Instead you should unwrap the error in the `else`

Reply:
Done